dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -283,31 +259,23 @@ def self.teacher_query(columns = '*')
     SQL
   end
 
-  # @param source [String] Source from which we want to extract data (can be a dashboard table name, or subquery)
-  # @param email_column [String] Column in source table we want to insert ino the email column
-  # @param data_columns [Array] Columns we want reshaped into a single JSON object
-  # @param source_is_subquery [Boolean] (default false) Set to true if source is a subquery, rather than a table name
-  # @param source_name [String] (default nil) Name for source (non-nil if using a subquery)
+  # @param source_name [String] The source table contains data we want to extract
+  # @param select_query [String] Query to select data from the source table
+  # @param data_columns [String] Columns we want reshaped into a single JSON object
   # @return [String] A SQL statement to extract and reshape data from the source table.
-  def self.get_extraction_query(source, email_column, data_columns, source_is_subquery=false, source_name=nil)
-    if source_name.nil? && source_is_subquery
-      raise 'Source name required if source is a subquery'
-    end
-
-    source_name ||= "dashboard.#{source}"
-    wrapped_source = source_is_subquery ? "(#{source}) AS subquery" : source
-
+  def self.get_extraction_query(source_name, select_query, *data_columns)
     <<~SQL
-      INSERT INTO #{ContactRollupsRaw.table_name} (email, sources, data, data_updated_at, created_at, updated_at)
+      INSERT INTO #{ContactRollupsRaw.table_name}
+        (email, sources, data, data_updated_at, created_at, updated_at)
       SELECT
-        #{email_column},
+        email,
         '#{source_name}' AS sources,
         #{create_json_object(data_columns)} AS data,
```

Comment:
Is there a fix that needs to be made to `create_json_object` to cover a case where `*data_columns` is null (which I think doesn't exist currently)?

Reply:
Oh, you have a test that covers this case. I'm surprised this works, is `data_columns` an empty array if no arguments are provided?

Reply:
Correct, `data_columns` is an empty array if there is no argument. `create_json_object` will fail if the input is `nil`, which should not happen. 

This is a quick test with variable arguments
```ruby
def test(*args); puts args.class; puts args.length; end
test   # will print out "Array" and "0"
```